### PR TITLE
Core/Skinning: Autolearn skill of skinned mob if not already learned

### LINKED ISSUE #28888: Core/Skinning: Learn skill of skinned mob if not already learned
### Description

Citing wowhead: 
> Skinning is the one profession in Dragonflight you don't have to go to a trainer to learn, although you can do that if you wish to. If you know basic Skinning going into Dragonflight, all you have to do is skin something and you're learn Dragonflight Skinning

Ref https://www.wowhead.com/de/guide/professions/skinning/overview-leveling-dragonflight

### Expected behaviour

1. Learn classic skinning
2. Skin outland mob
3. Automatically learn outland skinning with skill value 2

### Steps to reproduce the problem

1. Learn classic skinning
2. Skin outland mob
3. Nothing skill related happens

### Branch

master

### TC rev. hash/commit

229a7ac79d38b9db76387f92e846c04c151ad3aa

### Operating system

Win10

### Custom changes

none

### PR DESCRIPTION
**Pull request by meji46**
**Changes proposed:**

-  Autolearn skinning skill of skinned mob if not already learned (Legion still uses the old method looting item 129860)
> Skinning is the one profession in Dragonflight you don't have to go to a trainer to learn, although you can do that if you wish to. If you know basic Skinning going into Dragonflight, all you have to do is skin something and you're learn Dragonflight Skinning   

**Issues addressed:**
Closes #28888

**Tests performed:**
Builds, tested in-game.

**Known issues and TODO list:** (add/remove lines as needed)
None

### CODE REVIEW

**Review by mdX7 — APPROVED** [review]

**Comment by Shauren** [comment]
There is only one skinning spell, create a spell script instead

**Comment by meji46** [comment]
> There is only one skinning spell, create a spell script instead

What about lock spells for skinning gos?

**Comment by Shauren** [comment]
They don't grant teach you the skill (tested on PTR)

File: src/server/scripts/Spells/spell_generic.cpp [hunk 1]
```diff
@@ -5228,6 +5228,92 @@ class spell_gen_despawn_all_summons_owned_by_caster : public SpellScript
...
+    }
+
+    void HandleSkinningEffect(SpellEffIndex /*effIndex*/)
+    {
+        Player* player = GetCaster()->ToPlayer();
+        if (!player)
+            return;
+
+        Creature* creature = GetHitCreature();
+        if (!creature)
```
Comment (mdX7) [inline src/server/scripts/Spells/spell_generic.cpp:5274]: you can drop this check and use `GetHitUnit()->GetContentTuning()` for contentTuning instead without doing unit checks, as the target type is unit only according to db2 `TARGET_UNIT_TARGET_ANY`

**Pull request merged**